src/compile/util.ts

```diff
@@ -52,3 +55,84 @@ export function applyMarkConfig(marksProperties, model: Model, propsList: string
     }
   });
 }
+
+
+/**
+ * Builds an object with format and formatType properties.
+ *
+ * @param format explicitly specified format
+ */
+export function formatMixins(model: Model, channel: Channel, format: string) {
+  const fieldDef = model.fieldDef(channel);
+
+  let def: any = {};
+
+  switch (fieldDef.type) {
+    case QUANTITATIVE:
+      def.formatType = 'number';
+      break;
+    case TEMPORAL:
+      def.formatType = 'time';
+      break;
+  }
+
+  if (format !== undefined) {
+    def.format = format;
+  } else {
+    switch (fieldDef.type) {
+      case QUANTITATIVE:
+        def.format = model.numberFormat(channel);
+        break;
+      case TEMPORAL:
+        def.format = timeFormat(model, channel) || model.config().timeFormat;
+        break;
+    }
+  }
+
+  if (def.formatType && channel === TEXT) {
+    // text does not support format and formatType
+    // https://github.com/vega/vega/issues/505
+
+    const filter = def.formatType + (def.format ? ':\'' + def.format + '\'' : '');
+    return {
+      text: {
+        template: '{{' + model.field(channel, { datum: true }) + ' | ' + filter + '}}'
+      }
+    };
+  }
+
+  if (def.formatType === 'number') {
```

Comment:
Still because of line 92. I can drop this but it makes the logic hard to change.